Apps/CesiumViewer/CesiumViewer.js

```diff
@@ -53,20 +53,27 @@ function main() {
   let viewer;
   try {
     const hasBaseLayerPicker = !defined(imageryProvider);
+
+    let terrainProvider;
+    if (hasBaseLayerPicker) {
+      terrainProvider = createWorldTerrainAsync({
+        requestWaterMask: true,
+        requestVertexNormals: true,
+      });
+    }
+
     viewer = new Viewer("cesiumContainer", {
       imageryProvider: imageryProvider,
       baseLayerPicker: hasBaseLayerPicker,
       scene3DOnly: endUserOptions.scene3DOnly,
       requestRenderMode: true,
+      terrainProvider: terrainProvider,
     });
 
     if (hasBaseLayerPicker) {
-      const viewModel = viewer.baseLayerPicker.viewModel;
-      viewModel.selectedTerrain = viewModel.terrainProviderViewModels[1];
-    } else {
-      viewer.terrainProvider = createWorldTerrain({
-        requestWaterMask: true,
-        requestVertexNormals: true,
+      Promise.resolve(terrainProvider).then(() => {
```

Comment:
Note a huge deal, but it would be cleaner here to mark `main` as async and just use await here. Otherwise this is another unhandled promise that wouldn't end up in the below `catch` on error.